semantic-python/src/Language/Python/Core.hs

```diff
@@ -32,21 +35,44 @@ type CoreSyntax sig t = ( Member Core sig
 
 class Compile py where
   -- FIXME: we should really try not to fail
-  compile :: (CoreSyntax sig t, MonadFail m) => py -> m (t Name)
+  compile :: ( CoreSyntax syn t
+             , Member (Reader Path.RelFile) sig
```

Comment:
I like how descriptive this constraint is compared to a `Reader FilePath`, which could be anything (current working dir, root dir, the file itself, etc.)